build: Add systemd unit files

jcvenegas opened the pull request:
This patch add systemd unit files.

* Change the binary name to cc-agent.
* Generates a systemd service file if systemd is installed.
* Install systemd service and target if systemd is installed.

The new systemd target can be used to boot using this agent.
Fixes: #11 
Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

jcvenegas commented:
@sboeuf hi could you review this PR, I am using this script intall the agent 

https://gist.github.com/jcvenegas/5160e54294237342b4e79c34a7c9195b

mcastelino commented on clear-containers.service.in:
```diff
@@ -0,0 +1,10 @@
+Documentation=https://github.com/clearcontainers/agent
+
+[Service]
+ExecStart=@bindir@/@ccagent@
```
@jcvenegas @sboeuf do you not need to restart on failure? In the past we used to reboot on hyperstart crash. How will we handle agent crash

sboeuf replied:
We don't do anything for that for now, but I agree this is something we want.

mcastelino replied:
@sboeuf it would be great not to restart.

In our existing agent we have https://github.com/01org/cc-oci-runtime/blob/master/data/cc-agent.service#L9
FailureAction=poweroff

This makes debugging certain set of failures very hard if not impossible. But I do not know a better way to handle it, so that we can turn this off selectively.

jcvenegas committed:
build: Add systemd unit files

This patch add systemd unit files.

* Change the binary name to cc-agent.
* Generates a systemd service file if systemd is installed.
* Install systemd service and target if systemd is installed.

The new systemd target can be used to boot using this agent.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

sboeuf commented:
@mcastelino the poweroff option is the reason why the VM is shut down when hyperstart exits, right ?
I don't get what do you prefer ?

I can make this agent working for both 2.1 and 3.0 if we add this FailureAction option.

But when you say it makes the debugging hard, does that mean that you want to revert this option ? Or only for debug purpose ?

mcastelino commented:
@sboeuf yes. At least in debug mode we should disable the shutdown somehow without requiring to rebuild the image. Or at least provide a way where it does not shutdown.
